### ISSUE TITLE
Unwanted Upmix of sound source.

### ISSUE THREAD

**Issue by Choum28** [dosbox-staging/dosbox-staging#751]
**Describe the bug**
When the host machine is using a speakers configuration with surround speakers like 4.1, 5.1 and more (tested with analog connection). The stereo output of dosbox will be upmix which is the problem as sounds will be duplicate on front and rear speakers.

Example on 5.1 speakers with a dosgame which is stereo only.
Left sound will be play on : front left and rear right speakers instead of front left speaker only.
Right sound will be play on : front right and rear right speakers instead of front left speaker only.
Center sound will be play on all speakers at the same time.

**To Reproduce**
Have a pc with 5.1 sound configuration and speakers.
Launch the setmain.Exe of a game like duke nukem 3d.
Launch the sound FX test which will test left / center / right speakers.

**Expected behaviour**
Dosbox should ouput sound only to front left, front right speakers and eventually center speakers (to simulate center sound).
Look like dosbox is negociating corrrectly Stereo output but keep upmixing sound (see captures below).
Surround speakers should never be used (Exept if it exist some Dos game with Dolby Pro-Logic support)

**Environment (please complete the following information):**
- Windows 10x64 pro 19042
- Creative Sound Blaster AE7 (drivers : 6.0.105.55)
- 5.1 Speakers Logitech Z906 (analog connection : 3*mini-jack)

**Version of dosbox-staging:**
- Problem present on Dosbox Staging 0.76
- Also present on v0.77.0-alpha-6-g66393

**Configuration parameters:**
I juste change the Gus parameters to true and the mididevice config to use a specific external device for music.

**Additional context**
- Official dosbox 0.74.3 doesn't have this problem.

Note from Krc on discord :
DOSBox as a whole has no provisions to handle anything but 2-channels, so my guess is something downstream from SDL is doubling up the audio.
For example if we negotiated 4-channel audio with SDL, then DOSBox would need to pack quad-sample audio frames; which we don't do.  So the only way all four (or 5) of your channels are getting filled with samples is for something to be duplicating our two-channel audio.

![unknown 1](https://user-images.githubusercontent.com/16866992/101237631-b1e0c400-36da-11eb-9e97-be9a46e20761.png)

**Comment by Choum28** [comment]
Add wav capture as asked.

Tested on DN3D setmain audio configuration.
First test : Gravis Ultrasound
Second one at the end : Sound blaster.
[dosbox_000.zip](https://github.com/dosbox-staging/dosbox-staging/files/5647311/dosbox_000.zip)

**Comment by nemo93** [comment]
Thanks @Choum28 for reporting back and for confirming the issue is still there. Reopening + changing label.

**Comment by Burrito78** [comment]
@nemo93 Do you have any idea how this can be a DOSBox Staging problem? Isn't that an issue of the OS/sound driver?

**Comment by nemo93** [comment]
@Burrito78 I'm afraid I'm not the one to be asked :) Adding @Choum28 for confirmation.

**Comment by Choum28** [comment]
It could be a sound drivers setting if you volontary set an upmixing options in your sound card.
but by default, the OS/drivers will not play a sound on your 

> @nemo93 Do you have any idea how this can be a DOSBox Staging problem? Isn't that an issue of the OS/sound driver?

In this case, it's not. If a game / software and your sound card is correctly set, real stereo source will will not play any sound on rears speakers.

Of course it could be in such case : 

- sound card that have a setting that activate automatic upmixing of stereo source (ex : creative labs have an optionnal SBX surround setting for that).
- Some software have built in Upmixing settings (ex : AIMP), which by default will enable upmixing from stereo source to 5/7.1.
 
In dosbox staging, the problem is this upmixing cannot be disable.

**Comment by johnnovak** [comment]
Old DOSBox uses SDL1 and we're using SDL2. SDL1 is a relic from the past, I guess surround setups were not that widespread back then so it does not upmix 

The upmixing of mono and stereo sources to all sorts of surround setups seem to be a hardcoded SDL2 behaviour. It gets quite complicated and AFAIK applications ha no control over it (stereo will be always upmixed). Some details: 

https://github.com/libsdl-org/SDL/issues/5860#issuecomment-1180873596

I agree this could be handy to be able to disable SDL2 side upmixing, maybe I'll ask the SDL2 guys. But apparently people are happy with their upmixing, lots of games use SDL2 so I guess most people don't care or even notice to pester them about it.

**Comment by kcgen** [comment]
This is not related to the software, but whenever I've setup multi-channel output, I use a simple channel on/off box, like these:

https://www.amazon.ca/Speaker-Selector/s?k=Speaker+Selector

They give you immediate control over each pair of speakers. 

And a separate thought: These are the kinds of issue often drive someone to figuring out enough of the code to fix it themselves.

 *"You, know, that damn project didn't lift a finger to fix my quad channel speaker problem, so I rolled up my sleeves and figured it out. Turns out it wasn't that bad, and now I'm chipping away at other annoyances!"* -- @Choum28 , future-year 2024
